service/history/workflow/timer_sequence.go

```diff
@@ -203,7 +203,10 @@ func (t *timerSequenceImpl) LoadAndSortActivityTimers() []TimerSequenceID {
 	activityTimers := make(TimerSequenceIDs, 0, len(pendingActivities)*4)
 
 	for _, activityInfo := range pendingActivities {
-
+		// skip activities that are paused
+		if activityInfo.Paused {
+			continue
+		}
```

Comment:
I think about it - no, I don't need it. 

GenerateActivityRetryTasks is protected
I add a Paused check if task refresher.
As for the GenerateActivityTasks - in all other cases activity can't be paused.

Note: this will change as we move to paused rules.